sdk/execution/execution.go

```diff
@@ -123,19 +123,20 @@ func (e *Execution) validateExecutionOutput(serviceHash hash.Hash, taskKey strin
 }
 
 // Execute executes a task tasKey with inputData and tags for service serviceID.
-func (e *Execution) Execute(serviceHash hash.Hash, taskKey string, inputData map[string]interface{}, tags []string) (executionHash hash.Hash, err error) {
-	s, err := e.service.Get(serviceHash)
+func (e *Execution) Execute(instanceHash hash.Hash, taskKey string, inputData map[string]interface{}, tags []string) (executionHash hash.Hash, err error) {
+	// a task should be executed only if task's service is running.
+	instances, err := e.instance.List(&instancesdk.Filter{InstanceHash: instanceHash})
```

Comment:
maybe something wrong with the merge but this can be changed to 
```suggestion
	instances, err := e.instance.Get(instanceHash)
```
and the test about le len of instances can be removed